core/indigo-core/reaction/src/reaction_multistep_detector.cpp

```diff
@@ -238,6 +227,430 @@ void ReactionMultistepDetector::createSummBlocks()
+                                return std::nullopt;
+
+                            if (it_oz2->second.count(section))
+                                return zone1;
+                        }
+                    }
+                }
+
+                // different zone types are not mergeable
+                if (_zones[zone1.value().first].zone_type != _zones[zone2.value().first].zone_type && comm_zones.empty())
+                    return std::nullopt;
+            }
+
+            if (!current_zone.has_value())
+                current_zone = zone1;
+        }
+        else if (!current_zone.has_value())
+            return std::nullopt;
+
+        auto zidx = current_zone.value().first;
+        // if molecules have different zones - check if they are mergeable
+        auto& hull1 = _components[mol_idx1].second;
+        auto& hull2 = _components[mol_idx2].second;
+
+        const auto& coords = _zones[zidx].origin_coordinates;
+        // check if both molecules are on the same zone continuation
+        switch (_zones[zidx].zone_type)
+        {
+        case ZoneType::EPlus:
+            if ((doesVerticalLineIntersectPolygon(coords[0].x, hull1) && doesVerticalLineIntersectPolygon(coords[0].x, hull2)) ||
+                (doesHorizontalLineIntersectPolygon(coords[0].x, hull1) && doesHorizontalLineIntersectPolygon(coords[0].x, hull2)))
+                return zone1 ? zone1 : current_zone;
+            break;
+        case ZoneType::EArrow:
+            if ((doesRayIntersectPolygon(coords[0], coords[1], hull1) && doesRayIntersectPolygon(coords[0], coords[1], hull2)) ||
+                (doesRayIntersectPolygon(coords[1], coords[0], hull1) && doesRayIntersectPolygon(coords[1], coords[0], hull2)))
+                return zone1 ? zone1 : current_zone;
+            break;
+        case ZoneType::EPathWay: {
+            auto c_it = coords.begin();
+            const Vec2f& head = *c_it++;
+            const Vec2f& spine_beg = *c_it++;
+            const Vec2f& spine_end = *c_it++;
+            Vec2f tail(spine_beg.x, head.y);
+            if (doesRayIntersectPolygon(tail, head, hull1) && doesRayIntersectPolygon(tail, head, hull2))
+                return zone1 ? zone1 : current_zone;
+            for (; c_it != coords.end(); ++c_it)
+            {
+                Vec2f tail_start(spine_end.x, c_it->y);
+                if (doesRayIntersectPolygon(tail_start, *c_it, hull1) && doesRayIntersectPolygon(tail_start, *c_it, hull2))
+                    return zone1 ? zone1 : current_zone;
+            }
+        }
+        break;
+        }
+    }
+    return std::nullopt;
+}
+
+bool ReactionMultistepDetector::checkForOppositeSections(ZoneType zt, const std::unordered_set<int>& sections1, const std::unordered_set<int>& sections2)
+{
+    for (int section1 : sections1)
+    {
+        if (zt == ZoneType::EPathWay && section1 > 1) // pathway has only 2 opposite agents sections
```

Comment:
As you use section index in such way I think better add constants like ```pathway_sections_count```, ```pathway_section_top```, ```pathway_section_bottom```, ```pathway_max_opposite_section_index``` for all types and use it is this function

Reply:
fixed